service/builder.go

```diff
@@ -63,33 +50,13 @@ func (b *AgentServiceBuilder) Build(ctx context.Context) (service.Service, conte
 	return newAgentService(ctx, logAgent, persister, persisterShutdownFunc)
 }
 
-// validateNonGlobConfig returns an error if the configFile pointer is a glob.
-// Return nil if the configFile pointer is nil or not a single file path
-func (b *AgentServiceBuilder) validateNonGlobConfig() error {
-	if b.configFile == nil {
-		return nil
-	}
-
-	matches, err := filepath.Glob(*b.configFile)
-	switch {
-	case err != nil:
-		return err
-	case len(matches) > 1:
-		return errors.New("glob config paths not supported")
-	default:
-		return nil
-	}
-}
-
 func (b *AgentServiceBuilder) buildAgent() (*agent.LogAgent, error) {
 	agentBuilder := agent.NewBuilder(b.logger)
 
-	if b.configFile != nil {
-		agentBuilder = agentBuilder.WithConfigFiles([]string{*b.configFile})
-	}
-
-	if b.pluginDir != nil {
-		agentBuilder = agentBuilder.WithPluginDir(*b.pluginDir)
+	if b.config != nil {
```

Comment:
What happens if config is nil? Can we successfully build?

Reply:
The agent builder will throw an error and fail to build. We could return early here, not sure what we prefer.

Reply:
^ this is specifically the underlying LogAgentBuilder that will throw the error, not our service builder, if that wasn't clear.

Reply:
I think moving the error up and making it more obvious makes sense, so I've done that.